### PR TITLE
config.guess: update to 2021-05-24

### PR DESCRIPTION
**Pull request by aparcar**
This script hasn't seen an update in multiple years, update it to the
latest version.

The full changelog is available in the upstream git:
https://git.savannah.gnu.org/git/config.git

Signed-off-by: Paul Spooren <mail@aparcar.org>

### CODE REVIEW

**Comment by adschm** [comment]
I'm not really sure, but I think prefix should be "scripts:"?

File: scripts/config.guess [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #! /bin/sh
 # Attempt to guess a canonical system name.
-#   Copyright 1992-2018 Free Software Foundation, Inc.
+#   Copyright 1992-2021 Free Software Foundation, Inc.
```
Comment (adschm) [inline scripts/config.guess:3]: Can we update somebody else Copyright?

File: scripts/config.guess [hunk 2]
```diff
@@ -130,16 +131,14 @@ if test -f /.attbin/uname ; then
...
-UNAME_RELEASE=`(uname -r) 2>/dev/null` || UNAME_RELEASE=unknown
-UNAME_SYSTEM=`(uname -s) 2>/dev/null`  || UNAME_SYSTEM=unknown
-UNAME_VERSION=`(uname -v) 2>/dev/null` || UNAME_VERSION=unknown
+UNAME_MACHINE=$( (uname -m) 2>/dev/null) || UNAME_MACHINE=unknown
+UNAME_RELEASE=$( (uname -r) 2>/dev/null) || UNAME_RELEASE=unknown
+UNAME_SYSTEM=$( (uname -s) 2>/dev/null) || UNAME_SYSTEM=unknown
+UNAME_VERSION=$( (uname -v) 2>/dev/null) || UNAME_VERSION=unknown
 
-case "$UNAME_SYSTEM" in
+case $UNAME_SYSTEM in
```
Comment (adschm) [inline scripts/config.guess:139]: Why?

File: scripts/config.guess [hunk 3]
```diff
@@ -148,24 +147,36 @@ Linux|GNU|GNU/*)
...
+	#else
+	#include <stdarg.h>
+	/* First heuristic to detect musl libc.  */
+	#ifdef __DEFINED_va_list
+	LIBC=musl
+	#endif
 	#endif
 	EOF
-	eval "`$CC_FOR_BUILD -E "$dummy.c" 2>/dev/null | grep '^LIBC' | sed 's, ,,g'`"
+	eval "$($CC_FOR_BUILD -E "$dummy.c" 2>/dev/null | grep '^LIBC' | sed 's, ,,g')"
```
Comment (adschm) [inline scripts/config.guess:160]: Technically, you get the "" for free with $(), so you might remove them.

File: scripts/config.guess [hunk 4]
```diff
@@ -148,24 +147,36 @@ Linux|GNU|GNU/*)
...
+	if [ "$LIBC" = unknown ]; then
+		LIBC=gnu
 	fi
 	;;
 esac
 
 # Note: order is significant - the case branches are not exclusive.
 
-case "$UNAME_MACHINE:$UNAME_SYSTEM:$UNAME_RELEASE:$UNAME_VERSION" in
+case $UNAME_MACHINE:$UNAME_SYSTEM:$UNAME_RELEASE:$UNAME_VERSION in
```
Comment (adschm) [inline scripts/config.guess:179]: See above, why remove quotes on variables.

File: scripts/config.guess [hunk 5]
```diff
@@ -177,27 +188,27 @@ case "$UNAME_MACHINE:$UNAME_SYSTEM:$UNAME_RELEASE:$UNAME_VERSION" in
...
+		arch=$(echo "$UNAME_MACHINE_ARCH" | sed -e 's,^e\(armv[0-9]\).*$,\1,')
+		endian=$(echo "$UNAME_MACHINE_ARCH" | sed -ne 's,^.*\(eb\)$,\1,p')
 		machine="${arch}${endian}"-unknown
 		;;
 	    *) machine="$UNAME_MACHINE_ARCH"-unknown ;;
 	esac
 	# The Operating System including object format, if it has switched
 	# to ELF recently (or will in the future) and ABI.
-	case "$UNAME_MACHINE_ARCH" in
+	case $UNAME_MACHINE_ARCH in
```
Comment (adschm) [inline scripts/config.guess:211]: ...

File: scripts/config.guess [hunk 6]
```diff
@@ -531,11 +556,11 @@ EOF
...
 	# DG/UX returns AViiON for all architectures
-	UNAME_PROCESSOR=`/usr/bin/uname -p`
-	if [ "$UNAME_PROCESSOR" = mc88100 ] || [ "$UNAME_PROCESSOR" = mc88110 ]
+	UNAME_PROCESSOR=$(/usr/bin/uname -p)
+	if test "$UNAME_PROCESSOR" = mc88100 || test "$UNAME_PROCESSOR" = mc88110
 	then
-	    if [ "$TARGET_BINARY_INTERFACE"x = m88kdguxelfx ] || \
-	       [ "$TARGET_BINARY_INTERFACE"x = x ]
+	    if test "$TARGET_BINARY_INTERFACE"x = m88kdguxelfx || \
+	       test "$TARGET_BINARY_INTERFACE"x = x
```
Comment (adschm) [inline scripts/config.guess:563]: Why is "test" suddenly better than [] here?

**Comment by adschm** [comment]
Many of the changes look good, but I don't really understand why we need to remove quotes on variables and replace [] by test ...

**Comment by adschm** [comment]
> I just added the upstream files, see commit message. None of the changes are .add by me, i just copied the files.

Sorry, I wasn't aware of that. Maybe state explicitly in the commit message that you just copied the files.

Of course, under these circumstances I do not see a point in fixing "syntactical errors" in the files ...

**Comment by aparcar** [comment]
I updated the commit message and will merge this.

**Pull request closed without merging**